document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java

```diff
@@ -90,32 +105,45 @@ public boolean createCollection(String collectionName, Map<String, String> optio
                 + "%s TIMESTAMPTZ NOT NULL DEFAULT NOW(),"
                 + "%s TIMESTAMPTZ NOT NULL DEFAULT NOW()"
                 + ");",
-            collectionName, ID, DOCUMENT, CREATED_AT, UPDATED_AT);
+            identifier, ID, DOCUMENT, CREATED_AT, UPDATED_AT);
     try (PreparedStatement preparedStatement =
         client.getConnection().prepareStatement(createTableSQL)) {
       preparedStatement.executeUpdate();
     } catch (SQLException e) {
-      LOGGER.error("Exception creating table name: {}", collectionName);
+      LOGGER.error("Exception creating table name: {}", identifier);
       return false;
     }
     return true;
   }
 
+  private void createSchemaIfNotExists(@NonNull String schema) {
+    final String createSchemaSql = String.format("CREATE SCHEMA IF NOT EXISTS %s;", schema);
+    try (PreparedStatement preparedStatement =
+        client.getConnection().prepareStatement(createSchemaSql)) {
+      preparedStatement.execute();
+    } catch (SQLException e) {
+      LOGGER.error("Exception creating schema: {}", schema);
+    }
+  }
+
   @Override
   public boolean deleteCollection(String collectionName) {
-    String dropTableSQL = String.format("DROP TABLE IF EXISTS %s", collectionName);
+    PostgresTableIdentifier tableIdentifier = PostgresTableIdentifier.parse(collectionName);
+    String dropTableSQL = String.format("DROP TABLE IF EXISTS %s", tableIdentifier);
     try (PreparedStatement preparedStatement =
         client.getConnection().prepareStatement(dropTableSQL)) {
       int result = preparedStatement.executeUpdate();
       return result >= 0;
     } catch (SQLException e) {
-      LOGGER.error("Exception deleting table name: {}", collectionName);
+      LOGGER.error("Exception deleting table name: {}", tableIdentifier);
     }
     return false;
   }
 
   @Override
   public Collection getCollection(String collectionName) {
+    // FIXME - need to figure out listing schema tables before merging
```

Comment:
Is it still pending? I see that we included the collections with schema as part of listCollections?

Reply:
good catch! it's not, this is fixed now. will remove comment.